components/setting_item_max.tsx

```diff
@@ -135,27 +57,46 @@ export default class SettingItemMax extends React.PureComponent {
         document.removeEventListener('keydown', this.onKeyDown);
     }
 
-    onKeyDown = (e) => {
-        if (this.props.shiftEnter && e.keyCode === Constants.KeyCodes.ENTER && e.shiftKey) {
+    onKeyDown = (e: KeyboardEvent) => {
+        if (this.props.shiftEnter && e.key === Constants.KeyCodes.ENTER[0] && e.shiftKey) {
             return;
         }
-        if (this.props.disableEnterSubmit !== true && isKeyPressed(e, Constants.KeyCodes.ENTER) && this.props.submit && e.target.tagName !== 'SELECT' && e.target.parentElement && e.target.parentElement.className !== 'react-select__input' && !e.target.classList.contains('btn-cancel') && this.settingList.current && this.settingList.current.contains(e.target)) {
-            this.handleSubmit(e);
+        if (this.props.disableEnterSubmit !== true &&
+            isKeyPressed(e, Constants.KeyCodes.ENTER) &&
+            this.props.submit &&
+            e.target
+        ) {
+            const target = e.target as HTMLElement;
+            if (target.tagName !== 'SELECT' &&
+                target.parentElement &&
+                target.parentElement.className !== 'react-select__input' &&
+                !target.classList.contains('btn-cancel') &&
+                this.settingList.current &&
+                this.settingList.current.contains(target)
+            ) {
+                this.handleSubmit(e as any);
```

Comment:
please don't use `any` and instead properly type the event (which should not be needed, since it is of type `KeyboardEvent`, which should also be expected by the `handleSubmit` function)